src/WebJobs.Script/Description/DotNet/DirectSharedAssemblyProvider.cs

```diff
@@ -17,10 +18,12 @@ public DirectSharedAssemblyProvider(Assembly assembly)
             _assembly = assembly;
         }
 
-        public bool TryResolveAssembly(string assemblyName, AssemblyLoadContext targetContext, out Assembly assembly)
+        public bool TryResolveAssembly(string assemblyName, AssemblyLoadContext targetContext, ILogger logger, out Assembly assembly)
         {
             if (string.Compare(AssemblyNameCache.GetName(_assembly).Name, assemblyName, StringComparison.OrdinalIgnoreCase) == 0)
             {
+                logger.LogInformation($"{nameof(DirectSharedAssemblyProvider)} resolved shared assembly '{assemblyName}'");
```

Comment:
This will be fairly rare, right? It seems like `Debug`-level to me, but if it's not firing a bunch, this should work.

Reply:
Also, I'm curious what the category is for this ILogger? I'm not seeing where it comes from in this PR.

Reply:
The logger is passed here: https://github.com/Azure/azure-functions-host/pull/3849/files#diff-e51427694bc034bba6fa5a6b6d017957R170

I debated have this as debug, but kept it as info because it wasn't a frequent event. We can change that if needed.